.github/workflows/bump-chart-version.yaml

```diff
@@ -0,0 +1,355 @@
+                desiredParsed.patch += 1;
+              } else if (bumpType === 'minor-version') {
+                desiredParsed.minor += 1;
+                desiredParsed.patch = 0;
+              }
+              const desiredVersion = makeVersion(desiredParsed);
+              core.info(`Desired version for '${chartName}' is '${desiredVersion}'.`);
+
+              // If the Chart.yaml version is not what we expect, update it.
+              if (prVersion !== desiredVersion) {
+                core.info(`For '${chartName}', base was '${baseVersion}' but PR had '${prVersion}'. Changing version to '${desiredVersion}'.`);
+                const newChartContent = prContent.replace(
+                  /^version:\s+\S+/m,
+                  `version: ${desiredVersion}`
+                );
+
+                // Also replace file content locally so that helm-docs.sh script can properly update READMEs
+                fs.writeFileSync(`charts/${chartName}/Chart.yaml`, newChartContent, 'utf8');
+
+                fileChanges.push({
+                  path: `charts/${chartName}/Chart.yaml`,
+                  content: newChartContent
+                });
+                commitMessages.push(`bump version for ${chartName} to ${desiredVersion} (${bumpType})`);
+              } else {
+                core.info(`'${chartName}' version is already correct ('${prVersion}').`);
+              }
+
+              // Unless the bump type is no-version-bump, prepare CHANGELOG update.
+              if (bumpType === 'no-version-bump') {
+                core.info(`Skipping CHANGELOG update for '${chartName}' (no-version-bump).`);
+                continue;
+              }
+
+              // Get base and head CHANGELOG.md files.
+              let baseChangelog, prChangelog;
+              try {
+                baseChangelog = await github.rest.repos.getContent({
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                  path: `charts/${chartName}/CHANGELOG.md`,
+                  ref: mergeBaseSHA
+                });
+              } catch (error) {
+                core.setFailed(`Could not get base CHANGELOG.md for ${chartName}: ${error.message}`);
+                return;
+              }
+              try {
+                prChangelog = await github.rest.repos.getContent({
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                  path: `charts/${chartName}/CHANGELOG.md`,
+                  ref: pr.head.ref
+                });
+              } catch (error) {
+                core.setFailed(`Could not get head CHANGELOG.md for ${chartName}: ${error.message}`);
+                return;
+              }
+              if (prChangelog.data.sha !== baseChangelog.data.sha) {
+                core.info(`CHANGELOG.md for '${chartName}' has already been modified in this branch. Skipping changelog update.`);
+                continue;
+              }
```

Comment:
Known caveat: if changelog has already been updated, the changelog update will be skipped. 

We could try targeting only the version heading for update and leave any existing comments.

Reply:
Fixed in https://github.com/DataDog/helm-charts/pull/1785/commits/1fabad0defc59e8ee7d788156af355c99d81867e